### PR TITLE
Integrate protobuf (first part)

### PR DESCRIPTION
**Pull request by antoinealb**
This PR integrates some protobuf into the robot. It replaces *most* types exchanged on the messagebus by a protobuf generated one. It also pulls in some code that can encode protobuf messages for sending over UDP or saving to sdcard.

What will this be used for ? logging and debugging through generic bus introspection.

What is not done in this PR?

* Logging / debug itself
* Converting RobotState. I am afraid that this would cause the stack size to explose, and for now we can still have one non-protobuf data type laying around.
* Travis CI complains but I am not sure how to fix it yet.

I am particularly interested in feedback on naming stuff, as I am introducing a lot of data types.

### CODE REVIEW

File: master-firmware/src/base/encoder.c [hunk 1]
```diff
@@ -48,17 +49,9 @@ static THD_FUNCTION(encoders_thd, arg)
     chRegSetThreadName(__FUNCTION__);
 
     /* Setup and advertise encoders topic */
-    static messagebus_topic_t encoders_topic;
-    static MUTEX_DECL(encoders_topic_lock);
-    static CONDVAR_DECL(encoders_topic_condvar);
-    static encoders_msg_t encoders_topic_value;
+    static TOPIC_DECL(encoders_topic, EncodersPosition);
```
Comment (SyrianSpock) [inline master-firmware/src/base/encoder.c]: cool

File: master-firmware/src/base/encoder.c [hunk 2]
```diff
@@ -70,7 +63,7 @@ static THD_FUNCTION(encoders_thd, arg)
     setup_timer(STM32_TIM3);
 
     uint32_t left_old, right_old;
-    encoders_msg_t values = {0, 0};
+    EncodersPosition values = EncodersPosition_init_zero;
```
Comment (SyrianSpock) [inline master-firmware/src/base/encoder.c]: magic

Comment (antoinealb) [reply master-firmware/src/base/encoder.c]: wait until you see support for default values 👍

File: master-firmware/src/msgbus_protobuf.c [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+{
+    size_t offset;
+    int32_t max_len;
+
+    topic_metadata_t *metadata = (topic_metadata_t *)topic->metadata;
+
+    TopicHeader header;
+    MessageSize header_size, msg_size;
+
+    /* First populate information in the header */
```
Comment (SyrianSpock) [inline master-firmware/src/msgbus_protobuf.c]: could be split in sub functions to improve readability

File: master-firmware/protobuf/beacons.proto [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+message AlliedPosition
+{
+    option (nanopb_msgopt).msgid = 2;
+    required Timestamp timestamp = 1;
+    required float x = 2;
+    required float y = 3;
+}
+
+message BeaconSignal {
```
Comment (SyrianSpock) [inline master-firmware/protobuf/beacons.proto]: the more generic concept is `RangeBearing`: 
- range = distance
- bearing = angle
this covers more than our proximity beacon (e.g. lidar data).

Comment (antoinealb) [reply master-firmware/protobuf/beacons.proto]: So for range I can reuse the `Range` message, what do you think ?

**Pull request merged**